modin/core/dataframe/pandas/dataframe/dataframe.py

```diff
@@ -2202,12 +2202,44 @@ def map(
         PandasDataframe
             A new dataframe.
         """
-        map_fn = (
-            self._partition_mgr_cls.lazy_map_partitions
-            if lazy
-            else self._partition_mgr_cls.map_partitions
-        )
-        new_partitions = map_fn(self._partitions, func, func_args, func_kwargs)
+        if self.num_parts <= 1.5 * CpuCount.get():
+            map_fn = (
+                self._partition_mgr_cls.lazy_map_partitions
+                if lazy
+                else self._partition_mgr_cls.map_partitions
+            )
+            # old way
+            new_partitions = map_fn(self._partitions, func, func_args, func_kwargs)
+        else:
+            axis = (
+                1
+                if abs(self._partitions.shape[0] - CpuCount.get())
+                < abs(self._partitions.shape[1] - CpuCount.get())
+                else 0
+            )
```

Comment:
Please also add a comment why the axis is chosen this way.

```suggestion
            axis = 0
            if abs(self._partitions.shape[0] - CpuCount.get()) < abs(self._partitions.shape[1] - CpuCount.get()):
                axis = 1
```

Reply:
Ok, done